web/src/dropdown_widget.ts

```diff
@@ -98,10 +98,10 @@ export class DropdownWidget {
 
     constructor(options: DropdownWidgetOptions) {
         this.widget_name = options.widget_name;
-        this.widget_id = `#${CSS.escape(this.widget_name)}_widget`;
+        this.widget_selector = `#${CSS.escape(this.widget_name)}_widget`;
         // A widget wrapper may not exist based on the UI requirement.
-        this.widget_wrapper_id = `${this.widget_id}_wrapper`;
-        this.widget_value_selector = `${this.widget_id} .dropdown_widget_value`;
+        this.widget_wrapper_id = `${this.widget_selector}_wrapper`;
```

Comment:
This wrapper string also a selector, not an ID.

But constructing a selector using string concatenation like this is a sign that something dirty is going on. This construction won’t work with an arbitrary selector, only a selector that consists of `#` and an ID.

So it would be better to change the API to accept the ID (unescaped, without `#`), and then compute the selectors as `` `#${CSS.escape(widget_id)}` `` and `` `#${CSS.escape(`${widget_id}_wrapper`)}` ``.

Reply:
Thanks for feedback. I am tracking this, will get back to it later.